ratatui/src/widgets.rs

```diff
@@ -1,16 +1,513 @@
+//! - You must explicitly enable the feature flag to use them: `features = ["unstable-widget-ref"]`
+//! - They are not covered by semantic versioning guarantees until stabilized
+//!
+//! The traits are being evaluated for potential breaking changes and improvements. See the
+//! [tracking issue](https://github.com/ratatui/ratatui/issues/1287) for ongoing discussions and
+//! design considerations.
+//!
+//! # Rendering Widgets
+//!
+//! Widgets are typically rendered using the [`Frame`] type, which provides methods for rendering
+//! both consuming and reference-based widgets. These methods are usually called from the closure
+//! passed to [`Terminal::draw`].
+//!
+//! ## Rendering Consuming Widgets
+//!
+//! Most widgets in Ratatui are rendered using `Frame::render_widget()`, which consumes the widget
+//! when rendering. This is the standard approach for stateless widgets that don't need to persist
+//! data between frames.
+//!
+//! ```rust
+//! # use ratatui::{backend::TestBackend, Terminal};
+//! # use ratatui::widgets::Paragraph;
+//! # let backend = TestBackend::new(10, 3);
+//! # let mut terminal = Terminal::new(backend).unwrap();
+//! terminal.draw(|frame| {
+//!     let widget = Paragraph::new("Hello, world!");
+//!     frame.render_widget(widget, frame.area());
+//! });
+//! ```
+//!
+//! ## Rendering Widget References
+//!
+//! When you implement widgets on references (`Widget for &MyWidget`), you can render them directly
+//! using the same `Frame::render_widget()` method. This approach enables widget reuse without
+//! reconstruction and is the recommended pattern for new widgets.
+//!
+//! ```rust
+//! # use ratatui::{backend::TestBackend, Terminal};
+//! # use ratatui::widgets::{Block, Paragraph};
+//! # let backend = TestBackend::new(10, 3);
+//! # let mut terminal = Terminal::new(backend).unwrap();
+//! // Create the widget outside the draw closure
+//! let paragraph = Paragraph::new("Hello, world!").block(Block::bordered());
+//!
+//! terminal.draw(|frame| {
+//!     // Widget can be rendered by reference without being consumed
+//!     frame.render_widget(&paragraph, frame.area());
+//! });
+//!
+//! // The widget can be used again in subsequent frames
+//! terminal.draw(|frame| {
+//!     frame.render_widget(&paragraph, frame.area());
+//! });
+//! ```
+//!
+//! ## Rendering Stateful Widgets
+//!
+//! Widgets that need to maintain state between frames use `Frame::render_stateful_widget()`. This
+//! method takes both the widget and a mutable reference to its state, allowing the widget to read
+//! and modify state during rendering (such as updating scroll positions or handling selections).
+//!
+//! ```rust
+//! # use ratatui::{backend::TestBackend, Terminal};
+//! # use ratatui::widgets::{List, ListItem, ListState};
+//! # let backend = TestBackend::new(10, 3);
+//! # let mut terminal = Terminal::new(backend).unwrap();
+//! # let mut list_state = ListState::default();
```

Comment:
```suggestion
//! let mut list_state = ListState::default();
```

This is relevant to the types used, otherwise it could be confusing where or what `list_state` is.